Test/Integration/IntegrationTestBase.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Amwal\Payments\Test\Integration;
+
+use Amwal\Payments\Api\Data\RefIdDataInterface;
+use Amwal\Payments\Api\Data\RefIdDataInterfaceFactory;
+use Exception;
+use JsonException;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\App\Config\MutableScopeConfigInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Quote\Api\Data\CartItemInterfaceFactory;
+use Magento\Store\Model\ScopeInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\ObjectManager;
+use PHPUnit\Framework\TestCase;
+use TddWizard\Fixtures\Catalog\ProductBuilder;
+use TddWizard\Fixtures\Catalog\ProductFixture;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class IntegrationTestBase extends TestCase
+{
+    private const INTEGRATION_TEST_CONFIG = [
+        'currency/options/allow' => 'SAR',
+        'currency/options/base' => 'SAR',
+        'currency/options/default' => 'SAR',
+        'payment/amwal_payments/merchant_mode' => 'test',
+        'payment/amwal_payments/merchant_id_valid' => 1,
+        'payment/amwal_payments/merchant_id' => 'sandbox-amwal-e09ee380-d8c7-4710-a6ab-c9b39c7ffd47',
+        'payment/amwal_payments/integration_test_run' => 1
+    ];
+
+    protected const MOCK_PRODUCT_SKU = 'amwal_simple';
+
+    protected const MOCK_REF_ID_DATA = [
+        RefIdDataInterface::IDENTIFIER => '100',
+        RefIdDataInterface::CUSTOMER_ID => '0',
+        RefIdDataInterface::TIMESTAMP => '1707916143'
```

Comment:
if we can do one recording of the api request with one full successful transaction we can use the recorded payload for replaying the test, right?

Reply:
This is whats I tried initially, but the problem is that the reference ID is encrypted using the encryption key which is randomly generated with each test run, so the ref id would not match up and causes issues

Reply:
we can make it predictable by running using a preset integration set key each time and then using exactly the same key in our integration set run. the key for magento is stored unded `app/etc/env.php` like here
```
<?php
return [
    'backend' => [
        'frontName' => 'admin_1234'
    ],
    'remote_storage' => [
        'driver' => 'file'
    ],
    'queue' => [
        'consumers_wait_for_messages' => 1
    ],
    'crypt' => [
        'key' => 'ENCRYPTION_KEY'
    ],
```